src/sdk/namespace_fs.js

```diff
@@ -1127,6 +1131,25 @@ class NamespaceFS {
             throw err;
         }
     }
+
+    async _fail_if_archived_or_sparse_file(fs_account_config, file_path) {
+        try {
+            const stat = await nb_native().fs.stat(fs_account_config, file_path);
+            if (isDirectory(stat)) return;
+            // In order to verify if the file is stored in tape we compare sizes
+            // Multiple number of blocks by default block size and verify we get the size of the object
+            // If we get a size that is lower than the size of the object this means that it is taped or a spare file
+            // We had to use this logic since we do not have a POSIX call in order to verify that the file is taped
+            // This is why sparse files won't be accessible as well
+            if (stat.blocks * 512 < stat.size) {
+                dbg.log0(`validate_data_access: ${file_path} rejected`, stat);
+                throw new RpcError('INVALID_OBJECT_STATE', 'Attempted to access archived or sparse file');
+            }
+        } catch (err) {
+            dbg.error(`validate_data_access: ${file_path} error`, err);
+            throw this._translate_object_error_codes(err);
+        }
```

Comment:
Seems to me that this translate is redundant - this function is always called internally inside a try-catch that already calls translate. So I would remove this try catch inside this function altogether and let errors propagate.